packages/cli/test/integration/shared/testDb.ts

```diff
@@ -355,25 +357,31 @@ async function createExecution(
 	return execution;
 }
 
-/**
- * Store a successful execution in the DB and assign it to a workflow.
- */
-export async function createSuccessfulExecution(workflow: WorkflowEntity) {
-	return createExecution({ finished: true, status: 'success' }, workflow);
+// Test functions to create an execution of each status. 
+// Each execution is stored in the DB and assigned to a workflow
+
+/*
+* Create dummy execution for testing purposes.
+* The execution is stored in the DB and assigned to a workflow,
+* And is given a status to test filtering
+*/
+
+export async function createTestExecution(workflow: WorkflowEntity, status: ExecutionStatus) {
+	if (status == "success") {
+		return createExecution({finished: true, stoppedAt: new Date(), status: status}, workflow)
+	}
+	else if (status == "waiting") {
+		return createExecution({finished: false, waitTill: new Date(), status: status}, workflow)
+	}
+	return createExecution({finished: false, status: status}, workflow)
 }
```

Comment:
I was mostly worried about whether I should export `createExecution`, given that the previous code specifically avoided doing so. With the way I did it, I'd make it clear that this function is for testing only.

Reply:
ok i get it, then let only one way to create a test execution. Remove `createSuccessfulExecution` and `createErrorExecution`

It will maybe generate lot of changes in tests, but having only one way to create executions will make things easier